File: Upnp/ControlPoint/Services/Js/CpZappOrgTestWidget1.js
 

/**
* Service Proxy for zapp.org:TestWidget:1
* @module Zapp
* @title TestWidget
*/

var ServiceTestWidget = function(udn){	

	this.url = window.location.protocol + "//" + window.location.host + "/" + udn + "/zapp.org-TestWidget-1/control";  // upnp control url
	this.domain = "zapp.org";
	if (this.domain == "upnp.org") {
		this.domain = "schemas.upnp.org";
    }
	this.domain = this.domain.replace(/\./,"-");
	this.type = "TestWidget";
	this.version = "1";
	this.serviceName = "zapp.org-TestWidget-1";
	this.subscriptionId = "";  // Subscription identifier unique to each Subscription Manager 
	this.udn = udn;   // device name
	
	// Collection of service properties
	this.serviceProperties = {};
	this.serviceProperties["RegisterValue"] = new Zapp.ServiceProperty("RegisterValue");
	this.serviceProperties["RegisterIndex"] = new Zapp.ServiceProperty("RegisterIndex");
	this.serviceProperties["WidgetClass"] = new Zapp.ServiceProperty("WidgetClass");
	this.serviceProperties["ReadWriteRegister0"] = new Zapp.ServiceProperty("ReadWriteRegister0");
	this.serviceProperties["ReadWriteRegister1"] = new Zapp.ServiceProperty("ReadWriteRegister1");
	this.serviceProperties["ReadWriteRegister2"] = new Zapp.ServiceProperty("ReadWriteRegister2");
	this.serviceProperties["ReadWriteRegister3"] = new Zapp.ServiceProperty("ReadWriteRegister3");
	this.serviceProperties["ReadOnlyRegister4"] = new Zapp.ServiceProperty("ReadOnlyRegister4");
	this.serviceProperties["ReadOnlyRegister5"] = new Zapp.ServiceProperty("ReadOnlyRegister5");
	this.serviceProperties["ReadOnlyRegister6"] = new Zapp.ServiceProperty("ReadOnlyRegister6");
	this.serviceProperties["ReadOnlyRegister7"] = new Zapp.ServiceProperty("ReadOnlyRegister7");
}



/**
* Subscribes the service to the subscription manager to listen for property change events
* @method Subscribed
* @param {Function} serviceAddedFunction The function that executes once the subscription is successful
*/
ServiceTestWidget.prototype.subscribe = function (serviceAddedFunction) {
    Zapp.SubscriptionManager.addService(this,serviceAddedFunction);
}


/**
* Unsubscribes the service from the subscription manager to stop listening for property change events
* @method Subscribed
* @param {Function} serviceAddedFunction The function that executes once the subscription is successful
*/
ServiceTestWidget.prototype.unsubscribe = function () {
    Zapp.SubscriptionManager.removeService(this.subscriptionId);
}




/**
* Adds a listener to handle "RegisterValue" property change events
* @method RegisterValue_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.RegisterValue_Changed = function (stateChangedFunction) {
    this.serviceProperties.RegisterValue.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "RegisterIndex" property change events
* @method RegisterIndex_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.RegisterIndex_Changed = function (stateChangedFunction) {
    this.serviceProperties.RegisterIndex.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "WidgetClass" property change events
* @method WidgetClass_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.WidgetClass_Changed = function (stateChangedFunction) {
    this.serviceProperties.WidgetClass.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadWriteRegister0" property change events
* @method ReadWriteRegister0_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadWriteRegister0_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadWriteRegister0.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadWriteRegister1" property change events
* @method ReadWriteRegister1_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadWriteRegister1_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadWriteRegister1.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadWriteRegister2" property change events
* @method ReadWriteRegister2_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadWriteRegister2_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadWriteRegister2.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadWriteRegister3" property change events
* @method ReadWriteRegister3_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadWriteRegister3_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadWriteRegister3.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadOnlyRegister4" property change events
* @method ReadOnlyRegister4_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadOnlyRegister4_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadOnlyRegister4.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadOnlyRegister5" property change events
* @method ReadOnlyRegister5_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadOnlyRegister5_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadOnlyRegister5.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadOnlyRegister6" property change events
* @method ReadOnlyRegister6_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadOnlyRegister6_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadOnlyRegister6.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* Adds a listener to handle "ReadOnlyRegister7" property change events
* @method ReadOnlyRegister7_Changed
* @param {Function} stateChangedFunction The handler for state changes
*/
ServiceTestWidget.prototype.ReadOnlyRegister7_Changed = function (stateChangedFunction) {
    this.serviceProperties.ReadOnlyRegister7.addListener(function (state) 
	{ 
		stateChangedFunction(Zapp.SoapRequest.readIntParameter(state)); 
	});
}


/**
* A service action to SetReadWriteRegister
* @method SetReadWriteRegister
* @param {Int} RegisterIndex An action parameter
* @param {Int} RegisterValue An action parameter
* @param {Function} successFunction The function that is executed when the action has completed successfully
* @param {Function} errorFunction The function that is executed when the action has cause an error
*/
ServiceTestWidget.prototype.SetReadWriteRegister = function(RegisterIndex, RegisterValue, successFunction, errorFunction){	
	var request = new Zapp.SoapRequest("SetReadWriteRegister", this.url, this.domain, this.type, this.version);		
    request.writeIntParameter("RegisterIndex", RegisterIndex);
    request.writeIntParameter("RegisterValue", RegisterValue);
    request.send(function(result){
	
		if (successFunction){
			successFunction(result);
		}
	}, function(message, transport) {
		if (errorFunction) {errorFunction(message, transport);}
	});
}


/**
* A service action to GetWidgetClass
* @method GetWidgetClass
* @param {Function} successFunction The function that is executed when the action has completed successfully
* @param {Function} errorFunction The function that is executed when the action has cause an error
*/
ServiceTestWidget.prototype.GetWidgetClass = function(successFunction, errorFunction){	
	var request = new Zapp.SoapRequest("GetWidgetClass", this.url, this.domain, this.type, this.version);		
    request.send(function(result){
		result["WidgetClass"] = Zapp.SoapRequest.readIntParameter(result["WidgetClass"]);	
	
		if (successFunction){
			successFunction(result);
		}
	}, function(message, transport) {
		if (errorFunction) {errorFunction(message, transport);}
	});
}

File: Upnp/ControlPoint/Services/Cs/CpZappOrgTestWidget1.cs
using System;
using System.Runtime.InteropServices;
using System.Collections.Generic;
using System.Text;
using Zapp.Core;
using Zapp.ControlPoint;

namespace Zapp.ControlPoint.Proxies
{
    public interface ICpProxyZappOrgTestWidget1 : ICpProxy, IDisposable
    {
        void SyncSetReadWriteRegister(uint aRegisterIndex, uint aRegisterValue);
        void BeginSetReadWriteRegister(uint aRegisterIndex, uint aRegisterValue, CpProxy.CallbackAsyncComplete aCallback);
        void EndSetReadWriteRegister(IntPtr aAsyncHandle);
        void SyncGetWidgetClass(out uint aWidgetClass);
        void BeginGetWidgetClass(CpProxy.CallbackAsyncComplete aCallback);
        void EndGetWidgetClass(IntPtr aAsyncHandle, out uint aWidgetClass);
        void SetPropertyReadWriteRegister0Changed(CpProxy.CallbackPropertyChanged aReadWriteRegister0Changed);
        uint PropertyReadWriteRegister0();
        void SetPropertyReadWriteRegister1Changed(CpProxy.CallbackPropertyChanged aReadWriteRegister1Changed);
        uint PropertyReadWriteRegister1();
        void SetPropertyReadWriteRegister2Changed(CpProxy.CallbackPropertyChanged aReadWriteRegister2Changed);
        uint PropertyReadWriteRegister2();
        void SetPropertyReadWriteRegister3Changed(CpProxy.CallbackPropertyChanged aReadWriteRegister3Changed);
        uint PropertyReadWriteRegister3();
        void SetPropertyReadOnlyRegister4Changed(CpProxy.CallbackPropertyChanged aReadOnlyRegister4Changed);
        uint PropertyReadOnlyRegister4();
        void SetPropertyReadOnlyRegister5Changed(CpProxy.CallbackPropertyChanged aReadOnlyRegister5Changed);
        uint PropertyReadOnlyRegister5();
        void SetPropertyReadOnlyRegister6Changed(CpProxy.CallbackPropertyChanged aReadOnlyRegister6Changed);
        uint PropertyReadOnlyRegister6();
        void SetPropertyReadOnlyRegister7Changed(CpProxy.CallbackPropertyChanged aReadOnlyRegister7Changed);
        uint PropertyReadOnlyRegister7();
    }

    internal class SyncSetReadWriteRegisterZappOrgTestWidget1 : SyncProxyAction
    {
        private CpProxyZappOrgTestWidget1 iService;

        public SyncSetReadWriteRegisterZappOrgTestWidget1(CpProxyZappOrgTestWidget1 aProxy)
        {
            iService = aProxy;
        }
        protected override void CompleteRequest(IntPtr aAsyncHandle)
        {
            iService.EndSetReadWriteRegister(aAsyncHandle);
        }
    };

    internal class SyncGetWidgetClassZappOrgTestWidget1 : SyncProxyAction
    {
        private CpProxyZappOrgTestWidget1 iService;
        private uint iWidgetClass;

        public SyncGetWidgetClassZappOrgTestWidget1(CpProxyZappOrgTestWidget1 aProxy)
        {
            iService = aProxy;
        }
        public uint WidgetClass()
        {
            return iWidgetClass;
        }
        protected override void CompleteRequest(IntPtr aAsyncHandle)
        {
            iService.EndGetWidgetClass(aAsyncHandle, out iWidgetClass);
        }
    };

    /// <summary>
    /// Proxy for the zapp.org:TestWidget:1 UPnP service
    /// </summary>
    public class CpProxyZappOrgTestWidget1 : CpProxy, IDisposable, ICpProxyZappOrgTestWidget1
    {
        private Zapp.Core.Action iActionSetReadWriteRegister;
        private Zapp.Core.Action iActionGetWidgetClass;
        private PropertyUint iReadWriteRegister0;
        private PropertyUint iReadWriteRegister1;
        private PropertyUint iReadWriteRegister2;
        private PropertyUint iReadWriteRegister3;
        private PropertyUint iReadOnlyRegister4;
        private PropertyUint iReadOnlyRegister5;
        private PropertyUint iReadOnlyRegister6;
        private PropertyUint iReadOnlyRegister7;
        private CallbackPropertyChanged iReadWriteRegister0Changed;
        private CallbackPropertyChanged iReadWriteRegister1Changed;
        private CallbackPropertyChanged iReadWriteRegister2Changed;
        private CallbackPropertyChanged iReadWriteRegister3Changed;
        private CallbackPropertyChanged iReadOnlyRegister4Changed;
        private CallbackPropertyChanged iReadOnlyRegister5Changed;
        private CallbackPropertyChanged iReadOnlyRegister6Changed;
        private CallbackPropertyChanged iReadOnlyRegister7Changed;

        /// <summary>
        /// Constructor
        /// </summary>
        /// <remarks>Use CpProxy::[Un]Subscribe() to enable/disable querying of state variable and reporting of their changes.</remarks>
        /// <param name="aDevice">The device to use</param>
        public CpProxyZappOrgTestWidget1(CpDevice aDevice)
            : base("zapp-org", "TestWidget", 1, aDevice)
        {
            Zapp.Core.Parameter param;

            iActionSetReadWriteRegister = new Zapp.Core.Action("SetReadWriteRegister");
            param = new ParameterUint("RegisterIndex");
            iActionSetReadWriteRegister.AddInputParameter(param);
            param = new ParameterUint("RegisterValue");
            iActionSetReadWriteRegister.AddInputParameter(param);

            iActionGetWidgetClass = new Zapp.Core.Action("GetWidgetClass");
            param = new ParameterUint("WidgetClass");
            iActionGetWidgetClass.AddOutputParameter(param);

            iReadWriteRegister0 = new PropertyUint("ReadWriteRegister0", ReadWriteRegister0PropertyChanged);
            AddProperty(iReadWriteRegister0);
            iReadWriteRegister1 = new PropertyUint("ReadWriteRegister1", ReadWriteRegister1PropertyChanged);
            AddProperty(iReadWriteRegister1);
            iReadWriteRegister2 = new PropertyUint("ReadWriteRegister2", ReadWriteRegister2PropertyChanged);
            AddProperty(iReadWriteRegister2);
            iReadWriteRegister3 = new PropertyUint("ReadWriteRegister3", ReadWriteRegister3PropertyChanged);
            AddProperty(iReadWriteRegister3);
            iReadOnlyRegister4 = new PropertyUint("ReadOnlyRegister4", ReadOnlyRegister4PropertyChanged);
            AddProperty(iReadOnlyRegister4);
            iReadOnlyRegister5 = new PropertyUint("ReadOnlyRegister5", ReadOnlyRegister5PropertyChanged);
            AddProperty(iReadOnlyRegister5);
            iReadOnlyRegister6 = new PropertyUint("ReadOnlyRegister6", ReadOnlyRegister6PropertyChanged);
            AddProperty(iReadOnlyRegister6);
            iReadOnlyRegister7 = new PropertyUint("ReadOnlyRegister7", ReadOnlyRegister7PropertyChanged);
            AddProperty(iReadOnlyRegister7);
        }

        /// <summary>
        /// Invoke the action synchronously
        /// </summary>
        /// <remarks>Blocks until the action has been processed
        /// on the device and sets any output arguments</remarks>
        /// <param name="aRegisterIndex"></param>
        /// <param name="aRegisterValue"></param>
        public void SyncSetReadWriteRegister(uint aRegisterIndex, uint aRegisterValue)
        {
            SyncSetReadWriteRegisterZappOrgTestWidget1 sync = new SyncSetReadWriteRegisterZappOrgTestWidget1(this);
            BeginSetReadWriteRegister(aRegisterIndex, aRegisterValue, sync.AsyncComplete());
            sync.Wait();
            sync.ReportError();
        }

        /// <summary>
        /// Invoke the action asynchronously
        /// </summary>
        /// <remarks>Returns immediately and will run the client-specified callback when the action
        /// later completes.  Any output arguments can then be retrieved by calling
        /// EndSetReadWriteRegister().</remarks>
        /// <param name="aRegisterIndex"></param>
        /// <param name="aRegisterValue"></param>
        /// <param name="aCallback">Delegate to run when the action completes.
        /// This is guaranteed to be run but may indicate an error</param>
        public void BeginSetReadWriteRegister(uint aRegisterIndex, uint aRegisterValue, CallbackAsyncComplete aCallback)
        {
            Invocation invocation = iService.Invocation(iActionSetReadWriteRegister, aCallback);
            int inIndex = 0;
            invocation.AddInput(new ArgumentUint((ParameterUint)iActionSetReadWriteRegister.InputParameter(inIndex++), aRegisterIndex));
            invocation.AddInput(new ArgumentUint((ParameterUint)iActionSetReadWriteRegister.InputParameter(inIndex++), aRegisterValue));
            iService.InvokeAction(invocation);
        }

        /// <summary>
        /// Retrieve the output arguments from an asynchronously invoked action.
        /// </summary>
        /// <remarks>This may only be called from the callback set in the above Begin function.</remarks>
        /// <param name="aAsyncHandle">Argument passed to the delegate set in the above Begin function</param>
        public void EndSetReadWriteRegister(IntPtr aAsyncHandle)
        {
        }

        /// <summary>
        /// Invoke the action synchronously
        /// </summary>
        /// <remarks>Blocks until the action has been processed
        /// on the device and sets any output arguments</remarks>
        /// <param name="aWidgetClass"></param>
        public void SyncGetWidgetClass(out uint aWidgetClass)
        {
            SyncGetWidgetClassZappOrgTestWidget1 sync = new SyncGetWidgetClassZappOrgTestWidget1(this);
            BeginGetWidgetClass(sync.AsyncComplete());
            sync.Wait();
            sync.ReportError();
            aWidgetClass = sync.WidgetClass();
        }

        /// <summary>
        /// Invoke the action asynchronously
        /// </summary>
        /// <remarks>Returns immediately and will run the client-specified callback when the action
        /// later completes.  Any output arguments can then be retrieved by calling
        /// EndGetWidgetClass().</remarks>
        /// <param name="aCallback">Delegate to run when the action completes.
        /// This is guaranteed to be run but may indicate an error</param>
        public void BeginGetWidgetClass(CallbackAsyncComplete aCallback)
        {
            Invocation invocation = iService.Invocation(iActionGetWidgetClass, aCallback);
            int outIndex = 0;
            invocation.AddOutput(new ArgumentUint((ParameterUint)iActionGetWidgetClass.OutputParameter(outIndex++)));
            iService.InvokeAction(invocation);
        }

        /// <summary>
        /// Retrieve the output arguments from an asynchronously invoked action.
        /// </summary>
        /// <remarks>This may only be called from the callback set in the above Begin function.</remarks>
        /// <param name="aAsyncHandle">Argument passed to the delegate set in the above Begin function</param>
        /// <param name="aWidgetClass"></param>
        public void EndGetWidgetClass(IntPtr aAsyncHandle, out uint aWidgetClass)
        {
            uint index = 0;
            aWidgetClass = Invocation.OutputUint(aAsyncHandle, index++);
        }

        /// <summary>
        /// Set a delegate to be run when the ReadWriteRegister0 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadWriteRegister0Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadWriteRegister0Changed(CallbackPropertyChanged aReadWriteRegister0Changed)
        {
            lock (this)
            {
                iReadWriteRegister0Changed = aReadWriteRegister0Changed;
            }
        }

        private void ReadWriteRegister0PropertyChanged()
        {
            lock (this)
            {
                if (iReadWriteRegister0Changed != null)
                {
                    iReadWriteRegister0Changed();
                }
            }
        }

        /// <summary>
        /// Set a delegate to be run when the ReadWriteRegister1 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadWriteRegister1Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadWriteRegister1Changed(CallbackPropertyChanged aReadWriteRegister1Changed)
        {
            lock (this)
            {
                iReadWriteRegister1Changed = aReadWriteRegister1Changed;
            }
        }

        private void ReadWriteRegister1PropertyChanged()
        {
            lock (this)
            {
                if (iReadWriteRegister1Changed != null)
                {
                    iReadWriteRegister1Changed();
                }
            }
        }

        /// <summary>
        /// Set a delegate to be run when the ReadWriteRegister2 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadWriteRegister2Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadWriteRegister2Changed(CallbackPropertyChanged aReadWriteRegister2Changed)
        {
            lock (this)
            {
                iReadWriteRegister2Changed = aReadWriteRegister2Changed;
            }
        }

        private void ReadWriteRegister2PropertyChanged()
        {
            lock (this)
            {
                if (iReadWriteRegister2Changed != null)
                {
                    iReadWriteRegister2Changed();
                }
            }
        }

        /// <summary>
        /// Set a delegate to be run when the ReadWriteRegister3 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadWriteRegister3Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadWriteRegister3Changed(CallbackPropertyChanged aReadWriteRegister3Changed)
        {
            lock (this)
            {
                iReadWriteRegister3Changed = aReadWriteRegister3Changed;
            }
        }

        private void ReadWriteRegister3PropertyChanged()
        {
            lock (this)
            {
                if (iReadWriteRegister3Changed != null)
                {
                    iReadWriteRegister3Changed();
                }
            }
        }

        /// <summary>
        /// Set a delegate to be run when the ReadOnlyRegister4 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadOnlyRegister4Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadOnlyRegister4Changed(CallbackPropertyChanged aReadOnlyRegister4Changed)
        {
            lock (this)
            {
                iReadOnlyRegister4Changed = aReadOnlyRegister4Changed;
            }
        }

        private void ReadOnlyRegister4PropertyChanged()
        {
            lock (this)
            {
                if (iReadOnlyRegister4Changed != null)
                {
                    iReadOnlyRegister4Changed();
                }
            }
        }

        /// <summary>
        /// Set a delegate to be run when the ReadOnlyRegister5 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadOnlyRegister5Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadOnlyRegister5Changed(CallbackPropertyChanged aReadOnlyRegister5Changed)
        {
            lock (this)
            {
                iReadOnlyRegister5Changed = aReadOnlyRegister5Changed;
            }
        }

        private void ReadOnlyRegister5PropertyChanged()
        {
            lock (this)
            {
                if (iReadOnlyRegister5Changed != null)
                {
                    iReadOnlyRegister5Changed();
                }
            }
        }

        /// <summary>
        /// Set a delegate to be run when the ReadOnlyRegister6 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadOnlyRegister6Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadOnlyRegister6Changed(CallbackPropertyChanged aReadOnlyRegister6Changed)
        {
            lock (this)
            {
                iReadOnlyRegister6Changed = aReadOnlyRegister6Changed;
            }
        }

        private void ReadOnlyRegister6PropertyChanged()
        {
            lock (this)
            {
                if (iReadOnlyRegister6Changed != null)
                {
                    iReadOnlyRegister6Changed();
                }
            }
        }

        /// <summary>
        /// Set a delegate to be run when the ReadOnlyRegister7 state variable changes.
        /// </summary>
        /// <remarks>Callbacks may be run in different threads but callbacks for a
        /// CpProxyZappOrgTestWidget1 instance will not overlap.</remarks>
        /// <param name="aReadOnlyRegister7Changed">The delegate to run when the state variable changes</param>
        public void SetPropertyReadOnlyRegister7Changed(CallbackPropertyChanged aReadOnlyRegister7Changed)
        {
            lock (this)
            {
                iReadOnlyRegister7Changed = aReadOnlyRegister7Changed;
            }
        }

        private void ReadOnlyRegister7PropertyChanged()
        {
            lock (this)
            {
                if (iReadOnlyRegister7Changed != null)
                {
                    iReadOnlyRegister7Changed();
                }
            }
        }

        /// <summary>
        /// Query the value of the ReadWriteRegister0 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadWriteRegister0">Will be set to the value of the property</param>
        public uint PropertyReadWriteRegister0()
        {
            return iReadWriteRegister0.Value();
        }

        /// <summary>
        /// Query the value of the ReadWriteRegister1 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadWriteRegister1">Will be set to the value of the property</param>
        public uint PropertyReadWriteRegister1()
        {
            return iReadWriteRegister1.Value();
        }

        /// <summary>
        /// Query the value of the ReadWriteRegister2 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadWriteRegister2">Will be set to the value of the property</param>
        public uint PropertyReadWriteRegister2()
        {
            return iReadWriteRegister2.Value();
        }

        /// <summary>
        /// Query the value of the ReadWriteRegister3 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadWriteRegister3">Will be set to the value of the property</param>
        public uint PropertyReadWriteRegister3()
        {
            return iReadWriteRegister3.Value();
        }

        /// <summary>
        /// Query the value of the ReadOnlyRegister4 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadOnlyRegister4">Will be set to the value of the property</param>
        public uint PropertyReadOnlyRegister4()
        {
            return iReadOnlyRegister4.Value();
        }

        /// <summary>
        /// Query the value of the ReadOnlyRegister5 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadOnlyRegister5">Will be set to the value of the property</param>
        public uint PropertyReadOnlyRegister5()
        {
            return iReadOnlyRegister5.Value();
        }

        /// <summary>
        /// Query the value of the ReadOnlyRegister6 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadOnlyRegister6">Will be set to the value of the property</param>
        public uint PropertyReadOnlyRegister6()
        {
            return iReadOnlyRegister6.Value();
        }

        /// <summary>
        /// Query the value of the ReadOnlyRegister7 property.
        /// </summary>
        /// <remarks>This function is threadsafe and can only be called if Subscribe() has been
        /// called and a first eventing callback received more recently than any call
        /// to Unsubscribe().</remarks>
        /// <param name="aReadOnlyRegister7">Will be set to the value of the property</param>
        public uint PropertyReadOnlyRegister7()
        {
            return iReadOnlyRegister7.Value();
        }

        /// <summary>
        /// Must be called for each class instance.  Must be called before Core.Library.Close().
        /// </summary>
        public void Dispose()
        {
            DoDispose(true);
        }

        ~CpProxyZappOrgTestWidget1()
        {
            DoDispose(false);
        }

        private void DoDispose(bool aDisposing)
        {
            lock (this)
            {
                if (iHandle == IntPtr.Zero)
                {
                    return;
                }
                DisposeProxy();
                iHandle = IntPtr.Zero;
                iActionSetReadWriteRegister.Dispose();
                iActionGetWidgetClass.Dispose();
                iReadWriteRegister0.Dispose();
                iReadWriteRegister1.Dispose();
                iReadWriteRegister2.Dispose();
                iReadWriteRegister3.Dispose();
                iReadOnlyRegister4.Dispose();
                iReadOnlyRegister5.Dispose();
                iReadOnlyRegister6.Dispose();
                iReadOnlyRegister7.Dispose();
            }
            if (aDisposing)
            {
                GC.SuppressFinalize(this);
            }
        }
    }
}

